packages/tools/guiEditor/src/components/propertyTab/propertyGrids/gui/gridPropertyGridComponent.tsx

```diff
@@ -141,127 +155,87 @@ export class GridPropertyGridComponent extends React.Component<IGridPropertyGrid
+        }
+    }
+
+    isCloseTo(value: number, expected: number, epsilon: number) {
+        return Math.abs(value - expected) < epsilon;
+    }
+
+    adjustPercentages(definitions: string[], editFlags: boolean[]): string[] {
+        let percentageTotal = 0;
+        let modifiedEntriesPercentageTotal = 0;
+
+        const cellValues: string[] = [];
+
+        for (let i = 0; i < definitions.length; ++i) {
+            let value = definitions[i];
             const percent = this.checkPercentage(value);
-            if (this._rowEditFlags[i]) {
+            if (editFlags[i]) {
                 value = this.checkValue(value, percent);
-                if (percent) {
-                    editCount++;
-                }
             }
-
             if (percent) {
-                percentCount++;
-                const valueAsInt = parseInt(value.substring(0, value.length - 1));
-                total += valueAsInt / 100;
-                rowValues.push(valueAsInt / 100);
-            } else {
-                const valueAsInt = parseInt(value.substring(0, value.length - 2));
-                rowValues.push(valueAsInt);
+                percentageTotal += this.parsePercentage(value);
+                if (editFlags[i]) {
+                    modifiedEntriesPercentageTotal += this.parsePercentage(value);
+                }
             }
+            cellValues.push(value);
         }
 
-        const allEdited = editCount === percentCount;
+        let modifiedCellValues = [];
 
-        if (total > 1.0 || allEdited) {
-            const difference = total - 1.0;
-            const diff = Math.abs(difference);
-            for (let i = 0; i < grid.rowCount; ++i) {
-                if (this.checkPercentage(this._rowDefinitions[i])) {
-                    const value = rowValues[i];
-                    const weighted = diff * (value / total);
-                    rowValues[i] = difference > 0 ? value - weighted : value + weighted;
-                }
-            }
-        } else if (total < 1.0) {
-            const difference = 1.0 - total;
-            for (let i = 0; i < grid.rowCount; ++i) {
-                if (this.checkPercentage(this._rowDefinitions[i]) && this._rowEditFlags[i]) {
-                    const value = rowValues[i];
-                    total -= value;
-                }
-            }
-            for (let i = 0; i < grid.rowCount; ++i) {
-                if (this.checkPercentage(this._rowDefinitions[i]) && !this._rowEditFlags[i]) {
-                    const value = rowValues[i];
-                    const weighted = difference * (value / total);
-                    rowValues[i] = value + weighted;
+        // If the total percentage is not 100% we need to adjust the values based on the remaining percentage that was not modified by the user;
+        // If the remaining percentage is 0% we need to resize the modified entries to fit the remaining space
+        if (this.isCloseTo(percentageTotal, 1, 0.001)) {

```

Comment:
```suggestion
        if (this.isCloseTo(percentageTotal, 1)) {
```

(only if previous suggestion is fine :-) )